fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/connection/FHIRDbTenantDatasourceConnectionStrategy.java

```diff
@@ -54,40 +52,45 @@
 
     // number of nanoseconds in a millisecond
     private static final long NANOMS = 1000000;
-    
-    // JNDI address of the (proxy) datasource
-    private final String datasourceBaseName = "jdbc/fhir_";
+
+    // JNDI prefix  of the (proxy) datasource
+    private static final String DATASOURCE_BASE_NAME = "jdbc/fhir";
 
     // Cache of datasources we've found
     private final Map<String, DataSource> datasourceMap = new ConcurrentHashMap<>();
-    
+
     // the flavor of the database we are configured to represent
     private final FHIRDbFlavor flavor;
 
+    // Should we use read-only datasources for isReadOnly() requests?
+    private final boolean enableReadOnlyReplicas;
+
     /**
      * Public constructor. The proxy datasource must be present (registered in JNDI)
      * at server startup.
      * @throws FHIRPersistenceDBConnectException if the proxy datasource is not configured
      */
-    public FHIRDbTenantDatasourceConnectionStrategy(TransactionSynchronizationRegistry trxSyncRegistry, Action newConnectionAction) throws FHIRException {
+    public FHIRDbTenantDatasourceConnectionStrategy(TransactionSynchronizationRegistry trxSyncRegistry, Action newConnectionAction, boolean enableReadOnlyReplicas) throws FHIRException {
         super(trxSyncRegistry, newConnectionAction);
+        this.flavor = createFlavor();
+        this.enableReadOnlyReplicas = enableReadOnlyReplicas;
+    }
 
-        // Find the base JNDI name of the datasource we want to use
-        try {
-//            this.datasourceBaseName =
-//                    FHIRConfiguration.getInstance().loadConfiguration().getStringProperty(
-//                        FHIRConfiguration.PROPERTY_JDBC_DATASOURCE_JNDINAME, FHIRDbDAO.FHIRDB_JNDI_NAME_DEFAULT);
-            
-            if (log.isLoggable(Level.FINE)) {
-                log.fine("Using datasource JNDI name: " + datasourceBaseName);
-            }
-        } catch (Throwable e) {
-            FHIRException fx = new FHIRPersistenceDBConnectException("Failure acquiring datasource");
-            log.log(Level.SEVERE, fx.addProbeId("Failure to find proxy datasource in FHIR server configuration"), e);
-            throw fx;
+    public static String makeTenantDatasourceJNDIName(String jndiBase, String tenantId, String dsId, boolean readOnly) {
+        final StringBuilder builder = new StringBuilder();
+
+        builder.append(jndiBase);
+        builder.append("_");
+        builder.append(tenantId);
+        builder.append("_");
+        builder.append(dsId);
+
+        if (readOnly) {
+            // Allow the query to hit read-only replicas if we're supporting that
+            builder.append("_ro");
         }
-        
-        this.flavor = createFlavor();
+
+        return builder.toString();
     }
```

Comment:
I saw something in the documentation that we make a recommendation on the JNDI name but not required.  I think based on this it's required? correct?

Reply:
Ignore comment - line 130 answered.